mathics/core/symbols.py

```diff
@@ -474,8 +470,21 @@ def is_numeric(self, evaluation=None) -> bool:
             return symbol_definition.is_numeric
         return False
 
-    def default_format(self, evaluation, form) -> str:
-        return self.name
+    def is_uncertain_final_definitions(self, definitions) -> bool:
+        """
+        Used in Expression.do_format() to determine if we need to
+        (re)evaluate an expression.
+
+        Here, we have to be pessimistic and return True. For example,
+        in:
+
+           Context[]
+
+        this routine will get called where "self" is $System`Context. We
+        can't stop here, but must continue evaluation to get the function's value,
+        such as "Global`".
+        """
+        return True
```

Comment:
What about checking if ``definitions`` have a definition for  ``self.get_name()`` ?

Reply:
At this point in the code we don't know if the Mathics uses typed `Context[]` or `Context`. So I don't know if this routine has enough information to know whether to continue with an evaluation or not.

Reply:
Mathics uses `$Context` (or `Context[]`) for the current context,   and `Context[sym]` for the context of a symbol.

Reply:
But OK, if you want to start a `Definitions` object without the builtins, the best is your implementation: Symbols always should return True. Otherwise, we could have problems.

Reply:
So, I think this PR is ready to merge.

Reply:
Ok - we should keep this in mind. In this PR, the main focus was on clarity or describing what the situation currently is as clearly as I can. (And as you see, this is kind of a challenge).

When it comes to improving the code we should keep all of these ideas in mind and discuss them. It is possible in the interim of going over the rest of the code other changes will cause changes in how we approach improvements. 

On the other hand, I _have_ been making little improvements here and there. However hopefully these are not far reaching decisions. They have more been of the type: here we don't return False instead of True, or don't try to sort a list with less than 2 items.

Merging now, then.